src/main/scala/models/KeyWithContext.scala

```diff
@@ -0,0 +1,15 @@
+package CSVValidation
+case class KeyWithContext(
+    rowNumber: Long,
+    keyValues: List[Any],
+    var isDuplicate: Boolean = false
+) {
+
+  override def equals(obj: Any): Boolean =
```

Comment:
Would be good to put a comment explaining why we're overriding equals/hashCode in here.

Reply:
ok

Reply:
Best to use the [scaladoc](https://docs.scala-lang.org/style/scaladoc.html) style of comments. It *should* ensure that the test pops up when users use intellisense for help.

Reply:
okay